### PR TITLE
Darkmode toggler

### PR DESCRIPTION
**Pull request by Sristi2004-developer**
Created a ThemeContext.jsx file which provides  a DarkMode state to track whether the Dark mode is enabled and a toggleTheme function to switch between light and dark modes and used this context into the navbar component using useContext hook.

### CODE REVIEW

**Comment by Ruler45** [comment]
Dark mode doesn't persist after reload. FIX that

**Comment by Ecell23** [comment]
still the same issue. Take help from your friends in E-cell

File: src/context/ThemeContext.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+function ThemeContextProvider({ children }) {
+  let currentMode = localStorage.getItem("isDarkMode");
+  console.log(typeof currentMode);
+  if (currentMode == null) {
+    currentMode = "false";
+    localStorage.setItem("isDarkMode", "false");
+  } else {
+    if (currentMode == "false") {
+      document.body.classList.remove("dark-theme");
+      localStorage.setItem("isDarkMode", "false");
```
Comment (dheeraj1922d) [inline src/context/ThemeContext.jsx:17]: redundant local Storage update : updating local Storage multiple times within the render phase, which is unnecessary

File: src/context/ThemeContext.jsx
Comment (dheeraj1922d) [inline src/context/ThemeContext.jsx:1]: Use useEffect hook to update the document.body class and localStorage whenever isDarkMode changes

File: src/context/ThemeContext.jsx (hunk 1 above)
Comment (dheeraj1922d) [inline src/context/ThemeContext.jsx]: Redundant Local Storage Update : updating local Storage multiple times within the render phase, which is unnecessary

File: src/context/ThemeContext.jsx
Comment (dheeraj1922d) [inline src/context/ThemeContext.jsx:1]: Use UseEffect to update the document.body class and localStorage whenever isDarkMode changes.

**Review by dheeraj1922d — COMMENTED** [review]
Consider moving DOM manipulation and localStorage updates inside useEffect to prevent side effects during rendering.

**Pull request merged**